pkg/apis/catalog.cattle.io/v1/types.go

```diff
@@ -27,6 +27,9 @@ type RepoSpec struct {
 	// URL A http URL of the repo to connect to
 	URL string `json:"url,omitempty"`
 
+	// PlainHTTP is only valid for OCI URL's and allows insecure connections to registry without SSL check.
+	PlainHTTP bool `json:"plainHttp,omitempty"`
```

Comment:
It is a good practice to prefix insecure options to help users not sleep walk into it.:

```suggestion
	InsecurePlainHTTP bool `json:"plainHttp,omitempty"`
```

Reply:
@pjbgf In oras, there are two different things. I am using Oras library and so I am following their convention. 

Insecure - https://github.com/oras-project/oras/blob/main/cmd/oras/internal/option/remote.go#L107C20-L107C25
plain HTTP - https://github.com/oras-project/oras/blob/main/cmd/oras/internal/option/remote.go#L109

Reply:
I think the convention follows what curl does. See https://curl.se/docs/manpage.html#-k

Reply:
> In oras, there are two different things

Fair point, so let's keep it separately, but I would still add the `Insecure` prefix, to keep our Public API consistent.

Reply:
@pjbgf when do you think could be expected ETA for this to get merged and finally rolled out to release? This has caused a ton of people to not be able to update their install of Harbor for almost a year now.

Reply:
@johnwc This will be released in Rancher 2.9.0. @mattfarina can answer when will Rancher 2.9.0 will be out.

Reply:
@pjbgf Done.